### ISSUE TITLE
🏷️ Rename zinit annex repositories

### ISSUE THREAD

**Issue by pschmitt** [zdharma-continuum/zinit#38]
This sounds scarier than it is, promised!

To improve discoverability I suggest that we rename the repositories from `z-a-${annex_name}` to `zinit-annex-${annex_name}`.
This shouldn't break anything (™) as GitHub redirects to the new repo name.

- [x] https://github.com/zdharma-continuum/z-a-as-monitor -> zinit-annex-as-monitor
- [x] https://github.com/zdharma-continuum/z-a-bin-gem-node -> zinit-annex-bin-gem-node
- [x] https://github.com/zdharma-continuum/z-a-default-ice -> zinit-annex-default-ice
- [x] https://github.com/zdharma-continuum/z-a-man -> zinit-annex-man
- [x] https://github.com/zdharma-continuum/z-a-patch-dl - > zinit-annex-patch-dl
- [x] https://github.com/zdharma-continuum/z-a-rust -> zinit-annex-rust
- [x] https://github.com/zdharma-continuum/z-a-submods -> zinit-annex-submods
- [x] https://github.com/zdharma-continuum/z-a-test -> zinit-annex-test
- [x] https://github.com/zdharma-continuum/z-a-unscope -> zinit-annex-unscope

TODO:
- [x] Update the references to the various annexes in code
- [x] Update the references to the various annexes in the documentation

**Comment by alichtman** [comment]
TIL z-a- is short for zinit-annex. Yeah, definitely need this change.

**Comment by ryaminal** [comment]
In a similar vein, perhaps we rename the org to `zinit`?

**Comment by pschmitt** [comment]
I've articulated that I'm not exactly a fan of the current org given it's character length.

But "zinit" is a bit too narrow, there's way more stuff than zinit we manage here. It's the main project, that's true, but zdharma-continuum is a lot more than just zinit.

Also: rebranding now would make redo the work I did on most of the repos, but nothing a `find -type f -exec sed -i` can't solve.

We should discuss this further, but in another issue. This is a bit out of scope.
We should create a poll in the future, in good  zplugin/zinit tradition.
